server/src/main/java/org/elasticsearch/cluster/desirednodes/DesiredNodesSettingsValidator.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.cluster.desirednodes;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.cluster.metadata.DesiredNode;
+import org.elasticsearch.cluster.metadata.DesiredNodes;
+import org.elasticsearch.common.settings.ClusterSettings;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.core.Tuple;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+import static org.elasticsearch.common.util.concurrent.EsExecutors.NODE_PROCESSORS_SETTING;
+import static org.elasticsearch.common.util.concurrent.EsExecutors.createNodeProcessorsSetting;
+import static org.elasticsearch.node.Node.NODE_EXTERNAL_ID_SETTING;
+import static org.elasticsearch.node.Node.NODE_NAME_SETTING;
+
+public class DesiredNodesSettingsValidator {
+    private final ClusterSettings clusterSettings;
+
+    public DesiredNodesSettingsValidator(ClusterSettings clusterSettings) {
+        this.clusterSettings = clusterSettings;
+    }
+
+    public void validate(DesiredNodes desiredNodes) {
+        final List<Tuple<Integer, RuntimeException>> exceptions = new ArrayList<>();
+        final List<DesiredNode> nodes = desiredNodes.nodes();
+        for (int i = 0; i < nodes.size(); i++) {
+            final DesiredNode node = nodes.get(i);
+            try {
+                validate(node);
+            } catch (RuntimeException e) {
+                exceptions.add(Tuple.tuple(i, e));
+            }
+        }
+
+        if (exceptions.isEmpty() == false) {
+            final String nodeIndicesWithFailures = exceptions.stream()
+                .map(Tuple::v1)
+                .map(i -> Integer.toString(i))
+                .collect(Collectors.joining(","));
+            IllegalArgumentException invalidSettingsException = new IllegalArgumentException(
+                format(Locale.ROOT, "Nodes in positions [%s] contain invalid settings", nodeIndicesWithFailures)
+            );
+            for (Tuple<Integer, RuntimeException> exceptionTuple : exceptions) {
+                invalidSettingsException.addSuppressed(exceptionTuple.v2());
+            }
+            throw invalidSettingsException;
+        }
+    }
+
+    private void validate(DesiredNode node) {
+        if (node.version().before(Version.CURRENT)) {
```

Comment:
Should this be `Version.V_8_1_0`?

Reply:
This was intentional. The idea is that a cluster should not be able to go backwards, thus the next desired nodes should always be newer than current.